internal/bminventory/inventory.go

```diff
@@ -3043,6 +3042,39 @@ func handleReplyByType(params installer.PostStepReplyParams, b *bareMetalInvento
 	return err
 }
 
+func logReplyReceived(params installer.PostStepReplyParams, log *logrus.Entry) {
+	var sb strings.Builder
+	sb.WriteString(fmt.Sprintf("Received step reply <%s> from cluster <%s> host <%s> exit-code <%d>",
```

Comment:
Having the log in a separate function is good, but the usage of the string builder makes it more complicated than it should be.  I think it is better that the output is always written (even if it is empty), except for free-addresses at info level.

Reply:
right,  The condition not intended for removing an empty fields but for the log level and the free-addresses special behavior
Removing an empty data is a tiny improvement (We shouldn't write empty data)

Reply:
I think indication of empty data gives information.  Should not be removed